Object Store Get with a nats.Context causes a race condition

## Defect

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x] Included nats.go version
 - [x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats.go` and the `nats-server` if one was involved:
* `nats.go` version = `v1.27.0`
#### OS/Container environment:
mac
#### Steps or code to reproduce the issue:
Perform a `Get` on an `ObjectStore` while providing the `nats.Context` as a `GetObjectOpt` causes a race condition.

If we modify the `TestObjectBasics` test to add a `nats.Context(context.Background()` as an option to the seciont that gets the object back (line 95), the test will have a race condition.

```
func TestObjectBasics(t *testing.T) {
.
.
.
	// Now get the object back.
	result, err := obs.Get("BLOB", nats.Context(context.Background()))
	expectOk(t, err)
	expectOk(t, result.Error())
	defer result.Close()
.
.
.
```

#### Expected result:
The test to pass without any race conditions.

#### Actual result:
The test will pass when running without the `-race` flag, but as soon as we add the `-race` flag, then the test fails due to the race condition.

Race condition error:
```
==================
WARNING: DATA RACE
Read at 0x00c0001e56d0 by goroutine 41:
  github.com/nats-io/nats%2ego.(*obs).Get.func2()
      /Users/ajacques/git/nats.go/object.go:635 +0x209
  github.com/nats-io/nats%2ego.(*Conn).waitForMsgs()
      /Users/ajacques/git/nats.go/nats.go:3001 +0x782
  github.com/nats-io/nats%2ego.(*Conn).subscribeLocked.func1()
      /Users/ajacques/git/nats.go/nats.go:4232 +0x47

Previous write at 0x00c0001e56d0 by goroutine 6:
  github.com/nats-io/nats%2ego.(*obs).Get()
      /Users/ajacques/git/nats.go/object.go:667 +0xe31
  github.com/nats-io/nats.go/test.TestObjectBasics()
      /Users/ajacques/git/nats.go/test/object_test.go:101 +0xd78
  testing.tRunner()
      /usr/local/opt/go/libexec/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /usr/local/opt/go/libexec/src/testing/testing.go:1629 +0x47

Goroutine 41 (running) created at:
  github.com/nats-io/nats%2ego.(*Conn).subscribeLocked()
      /Users/ajacques/git/nats.go/nats.go:4232 +0x875
  github.com/nats-io/nats%2ego.(*Conn).subscribe()
      /Users/ajacques/git/nats.go/nats.go:4168 +0x144
  github.com/nats-io/nats%2ego.(*js).subscribe()
      /Users/ajacques/git/nats.go/js.go:1668 +0x1fca
  github.com/nats-io/nats%2ego.(*js).Subscribe()
      /Users/ajacques/git/nats.go/js.go:1250 +0xc5
  github.com/nats-io/nats%2ego.(*obs).Get()
      /Users/ajacques/git/nats.go/object.go:667 +0xe14
  github.com/nats-io/nats.go/test.TestObjectBasics()
      /Users/ajacques/git/nats.go/test/object_test.go:101 +0xd78
  testing.tRunner()
      /usr/local/opt/go/libexec/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /usr/local/opt/go/libexec/src/testing/testing.go:1629 +0x47

Goroutine 6 (running) created at:
  testing.(*T).Run()
      /usr/local/opt/go/libexec/src/testing/testing.go:1629 +0x805
  testing.runTests.func1()
      /usr/local/opt/go/libexec/src/testing/testing.go:2036 +0x8d
  testing.tRunner()
      /usr/local/opt/go/libexec/src/testing/testing.go:1576 +0x216
  testing.runTests()
      /usr/local/opt/go/libexec/src/testing/testing.go:2034 +0x87c
  testing.(*M).Run()
      /usr/local/opt/go/libexec/src/testing/testing.go:1906 +0xb44
  go.uber.org/goleak.VerifyTestMain()
      /Users/ajacques/go/pkg/mod/go.uber.org/goleak@v1.2.1/testmain.go:53 +0x70
  github.com/nats-io/nats.go/test.TestMain()
      /Users/ajacques/git/nats.go/test/main_test.go:23 +0x34c
  main.main()
      _testmain.go:740 +0x339
==================
--- FAIL: TestObjectBasics (0.85s)
    /Users/ajacques/git/nats.go/test/testing.go:1446: race detected during execution of test
FAIL
coverage: [no statements]
FAIL	github.com/nats-io/nats.go/test	1.416s
FAIL

```

--- a/object.go
+++ b/object.go
@@ -622,6 +622,7 @@ func (obs *obs) Get(name string, opts ...GetObjectOpt) (ObjectResult, error) {
 	result.digest = sha256.New()
 
 	processChunk := func(m *Msg) {
+		var err error
 		if ctx != nil {
 			select {
 			case <-ctx.Done():

--- a/test/object_test.go
+++ b/test/object_test.go
@@ -97,6 +97,12 @@ func TestObjectBasics(t *testing.T) {
 	expectOk(t, result.Error())
 	defer result.Close()
 
+	// Now get the object back with a context option.
+	result, err = obs.Get("BLOB", nats.Context(context.Background()))
+	expectOk(t, err)
+	expectOk(t, result.Error())
+	defer result.Close()
+
 	// Check info.
 	info, err = result.Info()
 	expectOk(t, err)

